share/github-backup-utils/ghe-backup-repositories

```diff
@@ -351,4 +353,13 @@ RULES
 done
 bm_end "$(basename $0) - Special Data Directories Sync"
 
+bm_start "$(basename $0) - Verifying Routes"
+
+cat $tempdir/*.rsync | sort | uniq > $tempdir/source_routes
+(cd $backup_dir/ && find * -mindepth 6 -maxdepth 7 -name \*.git -exec dirname {} \; | sort | uniq) > $tempdir/destination_routes
+
+diff $tempdir/source_routes $tempdir/destination_routes || echo "Warning: One or more repository networks and/or gists were not found on the source appliance. Please contact GitHub Enterprise Support for assistance."
```

Comment:
Several points:

- I think we should only show the difference output when a verbose backup is performed. 

- Do you think we should also force the backup to exit non-zero if this difference is found so it doesn't go completely under the radar? ie allow the rsync to continue but still "fail" the backup step.

- Is this going to have a major impact on the backup time, particularly for larger datasets?

Reply:
> I think we should only show the difference output when a verbose backup is performed.

As this information will be needed by Enterprise Support, I'd rather leave it in the standard output to prevent the need to re-run the backup with the `-v` flag. As this output forms part of the warning, I wouldn't consider it informational.

> Do you think we should also force the backup to exit non-zero if this difference is found so it doesn't go completely under the radar? ie allow the rsync to continue but still "fail" the backup step.

This seems like a sensible approach, and would certainly add additional visibility. I'll get this added in.

> Is this going to have a major impact on the backup time, particularly for larger datasets?

I don't expect it will have a significant impact, as I have tried to keep the comparison as lean and efficient as possible, but I do have a large backup archive here that I plan to use for some benchmarking.

Reply:
> This seems like a sensible approach, and would certainly add additional visibility. I'll get this added in.

On second thoughts, this probably isn't ideal, as it leaves the snapshot marked as incomplete, which will cause it to get pruned on the next successful backup.

Reply:
> On second thoughts, this probably isn't ideal, as it leaves the snapshot marked as incomplete, which will cause it to get pruned on the next successful backup.

Ah, good point.